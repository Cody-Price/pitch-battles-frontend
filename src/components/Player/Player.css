.player {
  transform: scaleX(-1);
  -webkit-transform: scaleX(-1);
  -moz-transform: scaleX(-1);
  -o-transform: scaleX(-1);

  position: absolute;
  height: 144px;
  width: 144px;
  top: 55px;
  left: 10px;
  image-rendering: pixelated;
  background-position: 0;
}

.player.idle {
  background-size: 576px 144px;
  background-position: 0;

  animation: player-four-frame infinite 1.3s steps(4);
  -webkit-animation: player-four-frame infinite 1.3s steps(4);
  -moz-animation: player-four-frame infinite 1.3s steps(4);
  -o-animation: player-four-frame infinite 1.3s steps(4);
}

.player.attack {
  background-size: 432px 144px;
  background-position: 0;

  animation: player-three-frame 1s steps(3);
  -webkit-animation: player-three-frame 1s steps(3);
  -moz-animation: player-three-frame 1s steps(3);
  -o-animation: player-three-frame 1s steps(3);
}

.player.victory {
  background-size: 432px 144px;

  animation: player-victory-frame 0.75s steps(3);
  -webkit-animation: player-victory-frame 0.75s steps(3);
  -moz-animation: player-victory-frame 0.75s steps(3);
  -o-animation: player-victory-frame 0.75s steps(3);

  background-position: -287px;
}

.player.hit {
  background-size: 864px 144px;
  background-position: 0;

  animation: player-six-frame 1s steps(6);
  -webkit-animation: player-six-frame 1s steps(6);
  -moz-animation: player-six-frame 1s steps(6);
  -o-animation: player-six-frame 1s steps(6);
}

.player.dead {
  background-size: 1008px 144px;

  animation: player-dead-frame 3s steps(7);
  -webkit-animation: player-dead-frame 3s steps(7);
  -moz-animation: player-dead-frame 3s steps(7);
  -o-animation: player-dead-frame 3s steps(7);
  background-position: -864px;
}

.hidden-preload {
  display: hidden;
}

/* -- PLAYER FOUR FRAME -- */

@keyframes player-four-frame {
  100% {
    background-position: -576px;
  }
}

@-webkit-keyframes player-four-frame {
  100% {
    background-position: -576px;
  }
}

@-moz-keyframes player-four-frame {
  100% {
    background-position: -576px;
  }
}

@-o-keyframes player-four-frame {
  100% {
    background-position: -576px;
  }
}

/* -- PLAYER THREE FRAME --  */

@keyframes player-three-frame {
  100% {
    background-position: -432px;
  }
}

@-webkit-keyframes player-three-frame {
  100% {
    background-position: -432px;
  }
}
@-moz-keyframes player-three-frame {
  100% {
    background-position: -432px;
  }
}

@-o-keyframes player-three-frame {
  100% {
    background-position: -432px;
  }
}

/* -- PLAYER SIX FRAME --  */

@keyframes player-six-frame {
  100% {
    background-position: -864px;
  }
}

@-webkit-keyframes player-six-frame {
  100% {
    background-position: -864px;
  }
}
@-moz-keyframes player-six-frame {
  100% {
    background-position: -864px;
  }
}

@-o-keyframes player-six-frame {
  100% {
    background-position: -864px;
  }
}

/* -- PLAYER DEAD FRAME --  */

@keyframes player-dead-frame {
  0% {
    background-position: 0;
  }
  100% {
    background-position: -1008px;
  }
}

@-webkit-keyframes player-dead-frame {
  0% {
    background-position: 0;
  }
  100% {
    background-position: -1008px;
  }
}
@-moz-keyframes player-dead-frame {
  0% {
    background-position: 0;
  }
  100% {
    background-position: -1008px;
  }
}

@-o-keyframes player-dead-frame {
  100% {
    background-position: -1008px;
  }
}

/* -- PLAYER THREE FRAME --  */

@keyframes player-victory-frame {
  0% {
    background-position: 0;
  }
  100% {
    background-position: -432px;
  }
}

@-webkit-keyframes player-victory-frame {
  0% {
    background-position: 0;
  }
  100% {
    background-position: -432px;
  }
}
@-moz-keyframes player-victory-frame {
  0% {
    background-position: 0;
  }
  100% {
    background-position: -432px;
  }
}

@-o-keyframes player-victory-frame {
  0% {
    background-position: 0;
  }
  100% {
    background-position: -432px;
  }
}
