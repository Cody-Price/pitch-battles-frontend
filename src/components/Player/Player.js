import React, { Component } from "react";
import avatars from "../../utilities/avatars";
import "./Player.css";
import "./CharacterAnims/Char1.css";
import "./CharacterAnims/Char2.css";
import "./CharacterAnims/Char3.css";
import "./CharacterAnims/Char4.css";
import "./CharacterAnims/Char5.css";
import "./CharacterAnims/Char6.css";
import "./CharacterAnims/Char7.css";
import "./CharacterAnims/Char8.css";
import "./CharacterAnims/Char9.css";
import "./CharacterAnims/Char10.css";

class Player extends Component {
  render() {
    return (
      <div
        className={`player ${avatars[this.props.avatar]} ${this.props.status}`}
      />
    );
  }
}

export default Player;
