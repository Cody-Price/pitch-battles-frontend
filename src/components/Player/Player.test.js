import React from "react";
import ReactDOM from "react-dom";
import Player from "./Player";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("Player", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Player />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
