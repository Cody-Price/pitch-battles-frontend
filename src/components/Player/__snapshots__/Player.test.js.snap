// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player should match the snapshot 1`] = `
<div
  className="player  undefined"
  style={
    Object {
      "backgroundImage": "url(./spritesheets/charundefined/undefined.png)",
    }
  }
>
  <div
    className="hidden-preload"
    style={
      Object {
        "backgroundImage": "url(./spritesheets/charundefined/attack.png)",
      }
    }
  />
  <div
    className="hidden-preload"
    style={
      Object {
        "backgroundImage": "url(./spritesheets/charundefined/idle.png)",
      }
    }
  />
  <div
    className="hidden-preload"
    style={
      Object {
        "backgroundImage": "url(./spritesheets/charundefined/hit.png)",
      }
    }
  />
  <div
    className="hidden-preload"
    style={
      Object {
        "backgroundImage": "url(./spritesheets/charundefined/dead.png)",
      }
    }
  />
  <div
    className="hidden-preload"
    style={
      Object {
        "backgroundImage": "url(./spritesheets/charundefined/victory.png)",
      }
    }
  />
</div>
`;
